pkg/awsutils/awsutils.go

```diff
@@ -56,7 +59,10 @@ const (
 	maxENIs   = 128
 	eniTagKey = "k8s-eni-key"
 
-	retryDeleteENIInternal = 5 * time.Second
+	exponentialBackoffMaxInterval    = 10 * time.Second
```

Comment:
I think some of these values need more thought put into them, and are potentially creating conflicting settings.

## Failure Cases

1. Transient Failure: network issue, something on AWS side, etc
2. Rate Limited: we are hitting their APIs too many times
3. AWS Outage: EC2 APIs are down for an extended period of time (minutes/hours)

### 1. Transient Failure
This should probably start working after a few seconds, all the defaults should be sufficient for retrying.

### 2. Rate Limited
This could be caused by a large amount of activity on our clusters (scaling up/down, adding lots of pods, etc).
This should resolve quite quickly as well, but without knowing more about AWS internals, it might be minutes.
Again leaving the defaults for the interval is probably quite sufficient, and we would probably want it to increase quite a bit if we are getting rate limited.

### 3. AWS Outage
This is the one I'm most worried about, as there is essentially no proper upper-bound, and it could result in the CNI plugin getting into an invalid state since there is no way to currently propagate an error back up and force it to retain the ENI that it was attempting to free.

In that case I kind of feel like the best option is a super long backoff (multi-hour?), but it might be best to resort to operator intervention in that case where we manually restart the ENI plugins in the case of an AWS issue.

## Suggested settings
Rather than setting the max interval or max retries, suggest only setting the max elapsed time, as that is really the main thing we care about (how long are we blocking other operations for).

By setting max interval, elapsed, time, and retries, you would only ever hit either max time, or max retries.